sgpt/cache.py

```diff
@@ -83,22 +86,32 @@ def __call__(self, func: Callable) -> Callable:
         :param func: The chat function to cache.
         :return: Wrapped function with chat caching.
         """
+
         def wrapper(*args, **kwargs):
-            chat_id = kwargs.pop("chat_id", None)
-            role = kwargs.pop("system", config.get("DEFAULT_SYSTEM_MESSAGE"))
+            # role stuff
+            role = kwargs.pop("role", None)
+            if not role:
```

Comment:
This `if not role:` syntax is necessary, because the auto *replacement if not exists* of `kwargs.pop` doesn't work because `'role'` will always be passed. (Even it what is passed is just a `none`.) 

It works the same way as chat_id.